playground/src/main/java/mx/yellowme/youst/playground/components/BarcodeDetector.kt

```diff
@@ -0,0 +1,103 @@
+package mx.yellowme.youst.playground.components
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.util.AttributeSet
+import android.widget.LinearLayout
+import com.google.android.gms.vision.Frame
+import com.google.android.gms.vision.barcode.Barcode
+import com.google.android.gms.vision.barcode.BarcodeDetector
+import io.fotoapparat.Fotoapparat
+import io.fotoapparat.parameter.ScaleType
+import io.fotoapparat.selector.back
+import kotlinx.android.synthetic.main.component_barcode_detector.view.*
+import mx.yellowme.youst.core.extensions.inflate
+import mx.yellowme.youst.playground.R
+
+interface BarcodeDelegate {
+
+    fun onPictureScanned(url: String)
+
+    fun onError(message: String)
+
+}
+
+/**
+ * @author adrianleyvasanchez
+ * @since 18,December,2019
+ */
+class BarcodeDetector @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    //region Attributes
+
+    private var detector: BarcodeDetector? = null
+
+    private var camera: Fotoapparat? = null
+
+    var delegate: BarcodeDelegate? = null
+
+    var barcodeFormat: Int = Barcode.QR_CODE
+
+    //endregion
+
+    //region Helpers
+
+    init {
+        inflate(R.layout.component_barcode_detector, context)
+        setup()
+    }
+
+    private fun setup() {
+        detector = BarcodeDetector.Builder(context)
+            .setBarcodeFormats(barcodeFormat)
+            .build()
+
+        if (!detector!!.isOperational) {
```

Comment:
Podemos evitar utilizar los asserts? 🤔 
También en vez de usar `!` podemos utilizar el .not()

Reply:
Creo que para evitar el force unwrapping podemos hacer que la variable `detector ` sea `lateinit`